src/main/java/net/dries007/tfc/api/types/ILivestockAI.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Work under Copyright. Licensed under the EUPL.
+ * See the project README.md and LICENSE.txt for more information.
+ */
+
+package net.dries007.tfc.api.types;
+
+
+import net.minecraft.entity.ai.EntityAIAvoidEntity;
+import net.minecraft.entity.ai.EntityAIEatGrass;
+import net.minecraft.entity.ai.EntityAIWanderAvoidWater;
+import net.minecraft.entity.ai.EntityAIWatchClosest;
+import net.minecraft.entity.ai.EntityAILookIdle;
+import net.minecraft.entity.ai.EntityAIMate;
+import net.minecraft.entity.ai.EntityAIPanic;
+import net.minecraft.entity.ai.EntityAISwimming;
+import net.minecraft.entity.ai.EntityAITempt;
+import net.minecraft.entity.passive.EntityAnimal;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+
+import net.minecraftforge.oredict.OreDictionary;
+
+import net.dries007.tfc.objects.entity.animal.EntityLionTFC;
+import net.dries007.tfc.objects.entity.animal.EntitySaberToothTFC;
+import net.dries007.tfc.objects.entity.animal.EntityWolfTFC;
+
+/**
+ * Use this to add AI Tasks that are the same between all TFC Livestock
+ * Does not apply to Vanilla derived mobs: Horse, Donkey, Mule, Llama, Camel
+ * Does apply to at least Pig, Cow, Chicken, Sheep, Goat, Duck, Alpaca,
+ * Used only in TFC worlds.
+ */
+public interface ILivestockAI extends ILivestock
```

Comment:
Thanks for review. Too bad IPredator can't be used by the Avoid Entity AI. :/
I'll look into making an Enum.

Reply:
That, or, just a static method somewhere. There's no point to implementing this interface - it's just a helper method to add common tasks to an entity. imo an enum doesn't make sense here particularly (it would work, as would this), but there's reason to think that some entities would want different combinations of behaviors.

`EntitiesTFC.addCommonLivestockAI(Entity entity)` would work fine.

Reply:
Done. Also incorporated all other TFC specific prey animal AI (not vanilla derived entities).
Is it really ok to have 9 EntityAIAvoidEntity tasks for the all the predators, plus EntityPlayer, and EntityMob?